core/internal/api/credentials.go

```diff
@@ -15,26 +21,31 @@ type CredentialProvider interface {
 	Apply(req *http.Request) error
 }
 
+// NewCredentialProvider creates a new credential provider based on the SDK
+// settings. Settings for JWT authentication are prioritized above API key
+// authentication
 func NewCredentialProvider(
 	settings *settings.Settings,
 ) (CredentialProvider, error) {
 	if settings.GetIdentityTokenFile() != "" {
-		return nil, fmt.Errorf("Identity federation via the wandb sdk " +
-			"is temporarily unavailable in wandb-core, or version 0.18.0 or " +
-			"later. Support for this feature will be reintroduced in an " +
-			"upcoming release. To continue using this feature, please " +
-			"downgrade to version 0.17.9 or lower using the following " +
-			"command: pip install wandb==0.17.9. Thank you for your patience.")
+		return NewOAuth2CredentialProvider(
+			settings.GetBaseURL(),
+			settings.GetIdentityTokenFile(),
+			settings.GetCredentialsFile(),
+		)
 	}
 	return NewAPIKeyCredentialProvider(settings)
 }
 
 var _ CredentialProvider = &apiKeyCredentialProvider{}
```

Comment:
why do you need this by the way?

Reply:
This is a GO convention to assert that the struct adheres to the CredentialProvider interface

Reply:
ah nice, thanks! i didn't know that... good to know